File: Spring Data/Hibernate Code First/Exercises/exercise_2/entities/Product.java
package exercise_2.entities;

import jakarta.persistence.*;

import java.util.Set;

@Entity
@Table(name = "products")
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Basic
    @Column(name = "name", nullable = false)
    private String name;

    @Basic
    @Column(name = "quantity", nullable = false)
    private Double quantity;

    @Basic
    @Column(name = "price", nullable = false)
    private Double price;

    @OneToMany(mappedBy = "product")
    private Set<Sale> sales;

    public Long getId() {
        return this.id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return this.name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Double getQuantity() {
        return this.quantity;
    }

    public void setQuantity(Double quantity) {
        this.quantity = quantity;
    }

    public Double getPrice() {
        return this.price;
    }

    public void setPrice(Double price) {
        this.price = price;
    }

    public Set<Sale> getSales() {
        return this.sales;
    }

    public void setSales(Set<Sale> sales) {
        this.sales = sales;
    }
}

File: C# OOP/Exam 2024-12-08/NetTraderSystem/Product.cs
namespace NetTraderSystem
{
    public class Product
    {
        public Product(string name, string category, double price)
        {
            Name = name;
            Category = category;
            Price = price;
        }

        public string Name { get; set; }
        public string Category { get; set; }
        public  double Price { get; set; }

        public override string ToString()
        {
            return $"Name: {Name}, Category: {Category} - ${Price:F2}";
        }
    }
}
